### PR TITLE
Always keep thread merged when there are platform views.

### PR DESCRIPTION
**Pull request by cyanglaz**
The unobstructed platform view has  been implemented. So there is a chance the overlay UIView can change even there is no changes in the platform view itself. Operations in overlay also needs to be run on the platform thread.
Because the current implementation depends on the "paint" traversal to know if an overlay is going to change; and we need to merge the thread before the "paint" traversal. There is no clean way to un-merged the threads when there are platform views.
This PR keeps the threads merged until the platform views are move out of the scene completely.
issue:
https://github.com/flutter/flutter/issues/56474

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 1]
```diff
@@ -252,6 +252,9 @@
 }
 
 bool FlutterPlatformViewsController::HasPendingViewOperations() {
+  // TODO(cyanglaz): https://github.com/flutter/flutter/issues/56474
```
Comment (blasten) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm:255]: `HasPendingViewOperations` isn't used, right?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h [hunk 2]
```diff
@@ -246,6 +246,12 @@ class FlutterPlatformViewsController {
   // have mutated for last layer tree.
   bool HasPendingViewOperations();
 
+  // Returns true if there are embedded in the scene at current frame
+  // Or there will be embedded in the next frame.
+  // TODO(cyanglaz): https://github.com/flutter/flutter/issues/56474
+  // Remove this method is the issue above is resloved.
```
Comment (blasten) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h:252]: Remove this method _when_ the issue above is _resolved_?.

**Comment by cyanglaz** [comment]
The `Mac Android AOT Engine` test was successful here https://chromium-swarm.appspot.com/task?id=4c232e09fb2fd410# in a manually triggered re-run.
Will land this PR even with this task appears red on github.

**Comment by xster** [comment]
(combing through recent untested PRs)
could we also add a FlutterPlatformViewsTest.m for this?

**Comment by cyanglaz** [comment]
AFAIK, we don't have a way to unit test this with current infra setup. (Maybe my knowledge is outdated) As far as integration tests, we probably can't test this behavior specifically. I will make sure to add an e2e test for the whole thread merging process before actually enabling the thread merging.

**Review by Afrogoddess31 — COMMENTED** [review]
Yup

**Pull request merged**